src/main/java/com/hotels/bdp/circustrain/bigquery/partition/PartitionValueFormatter.java

```diff
@@ -15,36 +15,54 @@
  */
 package com.hotels.bdp.circustrain.bigquery.partition;
 
-import java.util.List;
-import java.util.Set;
+import com.google.cloud.bigquery.FieldValue;
 
-import org.apache.hadoop.hive.metastore.api.FieldSchema;
+class PartitionValueFormatter {
 
-import com.google.common.collect.ImmutableSet;
+  private final String partitionKeyType;
+  private final FieldValue partitionFieldValue;
 
-class PartitionValueFormatter {
+  PartitionValueFormatter(FieldValue partitionFieldValue, String partitionKeyType) {
+    this.partitionFieldValue = partitionFieldValue;
+    this.partitionKeyType = partitionKeyType.toLowerCase();
+  }
 
-  private static final Set<String> NEEDS_FORMATTING = ImmutableSet.of("string");
-  private boolean format = false;
+  String formatValue() {
```

Comment:
You might as well just make this a static method and remove the constructor and all the state as I don't think we ever use this another way. So it will be more like `PartitionColumnFormatter`.